core/src/main/java/org/infinispan/factories/GlobalComponentRegistry.java

```diff
@@ -42,6 +30,17 @@
 import org.infinispan.util.logging.Log;
 import org.infinispan.util.logging.LogFactory;
 
+import javax.management.MBeanServer;
+import javax.management.MBeanServerFactory;
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
```

Comment:
Reorganizing imports makes the diff a bit difficult to read. This time it's ok, but please place reorgs in a separate commit next time.